### ISSUE TITLE
Feature request: Validate leading zero for numericality

### ISSUE THREAD

**Issue by mondwan** [ansman/validate.js#106]
# Goal:

* Expect `"010"` to be false in `numericality` validator

# Description:

It will be great if there is an option like `noLeadingZero` in `numericality`.

## Why

* Value from HTML5 number box is string
* The coercion of `+` will force `"010"` to be `10`

Facts above make `010` to be a valid number which it should not be in reality in my opinion.

# JS:

```javascript

var c = {
	"numericality": {
    "greaterThanOrEqualTo":1,
    "lessThanOrEqualTo":100,
    "message":"<id> from 1 to 100",
  },
  "presence": true,
};

// Expect to be false, but get true
console.log(
	'validation result ',
	validate.single('010', c)
);

// No problem
console.log(
	'validation result ',
	validate.single('10', c)
);
```

# Fiddler:

https://jsfiddle.net/8okbtbuz/2/

**Comment by ansman** [comment]
Agreed, though I think the option should be called `strict` instead.

**Comment by xhh2a** [comment]
Please make this an option, I would like cases where it is the current behavior.

**Comment by mondwan** [comment]
This option is optional by default

**Comment by mondwan** [comment]
By the way, option has been renamed to `strict` as requested in the pull request. However, I am not sure how to deal with floating point number. Should I allow all trailing zero if that is a float?

If yes, I can simply add a check and bypass this checking.

**Comment by ansman** [comment]
This is now released in 0.10.0
